ntb_hw_switchtec.c

```diff
@@ -191,6 +221,27 @@ static int switchtec_ntb_part_op(struct switchtec_ntb *sndev,
 	return -EIO;
 }
 
+static int switchtec_ntb_part_op(struct switchtec_ntb *sndev,
+				 struct ntb_ctrl_regs __iomem *ctl,
+				 u32 op, int wait_status)
+{
+	int rc;
+	int i = 0;
+
+	while (i++ < 10) {
+		rc = switchtec_ntb_part_op_no_retry(sndev, ctl, op,
+						    wait_status);
+		if (rc == -EAGAIN) {
+			msleep_interruptible(30);
```

Comment:
The return of msleep_interruptible() should be checked and return an error if it returns early (indicating the user hit ctrl-c during an operation initiated from userspace).

Alternatively just stick with regular msleep(). It doesn't seem like it's going to take very long so the user won't have to wait long if they do interrupt an operation.